Dataset sourced from kaggle site

https://github.com/najirh/Walmart_SQL_Python/blob/main/Walmart.csv

Import Libraries and Load Data

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Walmart.csv')
df.head()

invoice_id   Branch         City                category unit_price  \
0           1  WALM003  San Antonio       Health and beauty     $74.69   
1           2  WALM048    Harlingen  Electronic accessories     $15.28   
2           3  WALM067  Haltom City      Home and lifestyle     $46.33   
3           4  WALM064      Bedford       Health and beauty     $58.22   
4           5  WALM013       Irving       Sports and travel     $86.31   

   quantity      date      time payment_method  rating  profit_margin  
0       7.0  05/01/19  13:08:00        Ewallet     9.1           0.48  
1       5.0  08/03/19  10:29:00           Cash     9.6           0.48  
2       7.0  03/03/19  13:23:00    Credit card     7.4           0.33  
3       8.0  27/01/19  20:33:00        Ewallet     8.4           0.33  
4       7.0  08/02/19  10:37:00        Ewallet     5.3           0.48

Explore the Data

In [ ]:
df.shape

(10051, 11)

In [ ]:
df.describe()

invoice_id      quantity        rating  profit_margin
count  10051.000000  10020.000000  10051.000000   10051.000000
mean    5025.741220      2.353493      5.825659       0.393791
std     2901.174372      1.602658      1.763991       0.090669
min        1.000000      1.000000      3.000000       0.180000
25%     2513.500000      1.000000      4.000000       0.330000
50%     5026.000000      2.000000      6.000000       0.330000
75%     7538.500000      3.000000      7.000000       0.480000
max    10000.000000     10.000000     10.000000       0.570000

Terdapat masalah pada kolom quantity karena jumlahnya hanya 10020

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10051 entries, 0 to 10050
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_id      10051 non-null  int64  
 1   Branch          10051 non-null  object 
 2   City            10051 non-null  object 
 3   category        10051 non-null  object 
 4   unit_price      10020 non-null  object 
 5   quantity        10020 non-null  float64
 6   date            10051 non-null  object 
 7   time            10051 non-null  object 
 8   payment_method  10051 non-null  object 
 9   rating          10051 non-null  float64
 10  profit_margin   10051 non-null  float64
dtypes: float64(3), int64(1), object(7)
memory usage: 863.9+ KB


Kolom unit price dan quantity terdapat missing data

Data cleaning

In [ ]:
df.duplicated().sum()

51

Terdapat 51 data duplicate dalam data

In [ ]:
df.isnull().sum()

invoice_id         0
Branch             0
City               0
category           0
unit_price        31
quantity          31
date               0
time               0
payment_method     0
rating             0
profit_margin      0
dtype: int64

Terdapat 31 data null di kolom unit price dan quantity

In [ ]:
df.drop_duplicates(inplace=True)

In [ ]:
df.shape

(10000, 11)

In [ ]:
df.dropna(inplace=True)

In [ ]:
df.isnull().sum()

invoice_id        0
Branch            0
City              0
category          0
unit_price        0
quantity          0
date              0
time              0
payment_method    0
rating            0
profit_margin     0
dtype: int64

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9969 entries, 0 to 9999
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_id      9969 non-null   int64  
 1   Branch          9969 non-null   object 
 2   City            9969 non-null   object 
 3   category        9969 non-null   object 
 4   unit_price      9969 non-null   float64
 5   quantity        9969 non-null   float64
 6   date            9969 non-null   object 
 7   time            9969 non-null   object 
 8   payment_method  9969 non-null   object 
 9   rating          9969 non-null   float64
 10  profit_margin   9969 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 1.2+ MB


data null sudah tidak ada

In [ ]:
df.dtypes

invoice_id          int64
Branch             object
City               object
category           object
unit_price         object
quantity          float64
date               object
time               object
payment_method     object
rating            float64
profit_margin     float64
dtype: object

unit price perlu diubah menjadi tipe float

In [ ]:
df['unit_price'] = df['unit_price'].str.replace('$', '').astype(float)

In [ ]:
df.columns = df.columns.str.lower()

Feature Engineering

In [ ]:
df['total'] = df['unit_price'] * df['quantity']
df.head()

invoice_id   branch         city                category  unit_price  \
0           1  WALM003  San Antonio       Health and beauty       74.69   
1           2  WALM048    Harlingen  Electronic accessories       15.28   
2           3  WALM067  Haltom City      Home and lifestyle       46.33   
3           4  WALM064      Bedford       Health and beauty       58.22   
4           5  WALM013       Irving       Sports and travel       86.31   

   quantity      date      time payment_method  rating  profit_margin   total  
0       7.0  05/01/19  13:08:00        Ewallet     9.1           0.48  522.83  
1       5.0  08/03/19  10:29:00           Cash     9.6           0.48   76.40  
2       7.0  03/03/19  13:23:00    Credit card     7.4           0.33  324.31  
3       8.0  27/01/19  20:33:00        Ewallet     8.4           0.33  465.76  
4       7.0  08/02/19  10:37:00        Ewallet     5.3           0.48  604.17

Download Cleaned Data

In [ ]:
from google.colab import files
df.to_csv('walmart_clean.csv', index=False)
files.download('walmart_clean.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>